src/types/streaming_server/settings.rs

```diff
@@ -22,3 +23,112 @@ pub struct Settings {
     pub bt_download_speed_hard_limit: f64,
     pub bt_min_peers_for_stable: u64,
 }
+mod empty_string_as_null {
+    use serde::{self, Deserialize, Deserializer, Serializer};
+
+    pub fn serialize<S>(value: &Option<String>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *value {
+            None => serializer.serialize_str(""),
+            Some(ref s) => serializer.serialize_str(s),
+        }
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s: String = Deserialize::deserialize(deserializer)?;
+        if s.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(s))
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
Move the module inside `empty_string_as_null`
Also make sure to:
- [ ] use serde_josn::json to generate **valid** json instead of relying on strings like now.
- [ ] Instead of using Settings, just create your own structure in the test module, e.g.:
```rust
        #[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
        struct TestStruct {
            #[serde(with = "super")]
            field: Option<String>,
        }
```

You can still use the remote_https if you want but the behavior of the module is generic, just keep that in mind when writing the test cases, i.e. more generic name of the test of what's actually testing is a good rule of thumb.